### PR TITLE
fixes connection tests and prints config file used in test_db.py

### LINKED ISSUE #3038: db_test checking mysql connection suggested enhancement
Hi Nupic,
This is probably more of an enhancement suggestion than an issue.
I was having issues running swarms using my custom defined mysql credentials. The swarm was outputting database errors, however the recommended db_test.py was telling me I was successfully connecting.

I later realised that it was due to my custom mysql user being restricted to my project's database schema. A, now obvious, problem that I overlooked!

Perhaps the db_test.py could include tests to check that it has permissions to create the tables necessary for swarming or perhaps mention it in the documentation?
Thank you

### PR DESCRIPTION
**Pull request by andrewmalta13**
fixes #3038 

please review:
@rhyolight

### CODE REVIEW

File: examples/swarm/test_db.py [hunk 1]
```diff
@@ -19,41 +19,101 @@
...
+        conn = pymysql.connect(host=host, port=port, user=user, passwd=passwd)
+        cursor = conn.cursor()
+        cursor.execute('CREATE DATABASE IF NOT EXISTS testing')
+        conn.select_db('testing')
+        cursor.execute("CREATE TABLE test \
+                                (teststring VARCHAR(255),\
+                                 someint INT)")
+        cursor.execute("INSERT INTO test VALUES ('testing123', 123)")
+        cursor.execute('DROP TABLE IF EXISTS test')
+        cursor.execute('DROP DATABASE IF EXISTS testing')
```
Comment (rhyolight) [inline examples/swarm/test_db.py]: From [style guide](https://github.com/numenta/nupic/wiki/Python-Style-Guide#quotes); 

> By default, double quotes should be used for all strings.

Here and elsewhere throughout the file.

File: examples/swarm/test_db.py [hunk 2]
```diff
@@ -19,41 +19,101 @@
...
+        return True
+
+    except pymysql.err.OperationalError:
+        return False
+
+
+def db_validator():
+    """
+       let the user know what NuPIC config file is being used
```
Comment (rhyolight) [inline examples/swarm/test_db.py]: Nitpick: capitalize first word in sentence.

File: examples/swarm/test_db.py [hunk 3]
```diff
@@ -19,41 +19,101 @@
...
+
+
+    print "This script will validate that your MySQL is setup correctly for NuPIC."
+    print "MySQL is required for NuPIC swarming. The settings are defined in a "
+    print "configuration file found in $NUPIC/src/nupic/support/nupic-default.xml "
+    print "Out of the box those settings contain MySQL's default access "
+    print "credentials."
+    print
+    print "The nupic-default.xml can be duplicated to define user specific changes,"
+    print " calling the copied file $NUPIC/src/nupic/support/nupic-site.xml"
```
Comment (rhyolight) [inline examples/swarm/test_db.py]: Unnecessary whitespace at beginning of string.

**Comment by rhyolight** [comment]
@andrewmalta13 Left some nitpicks, but otherwise great work.

File: examples/swarm/test_db.py [hunk 4]
```diff
@@ -19,41 +19,103 @@
...
+    conn = pymysql.connect(host=host, port=port, user=user, passwd=passwd)
+    cursor = conn.cursor()
+    cursor.execute("CREATE DATABASE IF NOT EXISTS testing")
+    conn.select_db("testing")
+    cursor.execute("CREATE TABLE test \
+                              (teststring VARCHAR(255),\
+                               someint INT)")
+    cursor.execute("INSERT INTO test VALUES ('testing123', 123)")
+    cursor.execute("DROP TABLE IF EXISTS test")
+    cursor.execute("DROP DATABASE IF EXISTS testing")
```
Comment (rhyolight) [inline examples/swarm/test_db.py]: This is very much an edge-case, but what if a user has a database called `testing` that already exists? This script would drop it. I suggest renaming the test db to something more unique like `nupic_db_test`.

Comment (andrewmalta13) [inline examples/swarm/test_db.py]: I agree :), that would really piss someone off.

File: examples/swarm/test_db.py [hunk 5]
```diff
@@ -19,41 +19,103 @@
...
+  print "$NUPIC/src/nupic/support/nupic-site.xml Refer to the "
+  print "nupic-default.xml for additional instructions."
+  print
+  print "Defaults: localhost, 3306, root, no password"
+  print
+  print "Retrieved the following NuPIC configuration using: ", fileused
+  print "    host   :    ", host
+  print "    port   :    ", port
+  print "    user   :    ", user
+  print "    passwd :    ", passwd
```
Comment (rhyolight) [inline examples/swarm/test_db.py]: Damn, I should have noticed this before, but some people will not appreciate their plaintext password being printed to the console. I suggest you remove this line or replace it with `XXXXXXXX`.

**Comment by andrewmalta13** [comment]
@rhyolight replaced password with a number of *'s  based on the length of the password. So in the case that they don't have a password, the behavior is the same as before. What do you think?
